### ISSUE TITLE
What is the point of upgrading an extension that is not 2.0 compatible?

### ISSUE THREAD

**Issue by DonKolia** [EverMCServer/vscode-BetonQuest#7]
![image](https://github.com/EverMCServer/vscode-BetonQuest/assets/55564649/168685b1-b0b4-414d-b9d1-af7472850cae)

**Comment by k-jiang** [comment]
Please check out #5.

**Comment by DonKolia** [comment]
I have a idea. U can put a button here in the bottom. When u click, button add this code:
The button adds this code and the space to align the lines below it

conversations:
  NAME:

**Comment by DonKolia** [comment]
Maybe u need add spaces in the extension, u can locate the lines with spaces:
![image](https://github.com/EverMCServer/vscode-BetonQuest/assets/55564649/4fb4fb86-7fcf-4ff0-aecf-283fc21a865f)

**Comment by DonKolia** [comment]
And idk where i can edit the extension

**Comment by DonKolia** [comment]
Or u can remove the real-time sync editor --> yml, if u remove something in the editor, u need remove manual in the YML.
With this, you can add delimiters so that the extension does not delete any lines above those delimiters.

![image](https://github.com/EverMCServer/vscode-BetonQuest/assets/55564649/e6ff000b-f3db-4170-8107-8f6bfba193a9)

**Comment by DonKolia** [comment]
U answer me or what?

**Comment by DonKolia** [comment]
The tutorial: https://www.youtube.com/watch?v=iWwg_URQ-NY

**Comment by k-jiang** [comment]
This extension is still in a very primitive stage. The reason why I develop 1.0 support first is already explained in #5. That also why there are so many things not supported now, like the 2.0-DEV format.

Since the development on 2.0-DEV support is already began, I'm afraid I can not fulfill your request about the format conversion button. Such a feature will confuse the user when they accidentally saved the files as 2.0-DEV, where the project is actually 1.0. It will break a lot of things, like positioning the option box when click on the Yaml text. Additional works are introduced as well. For instance, I need to create a new loader for the new format, and dealing with other sections like "events" and "conditions" that allowed to be exist on the new format (if someone trying to import a 2.0-DEV file).

In summary such a feature might be buggy and very time-consuming. It is better to continue the development on the propper 2.0-DEV support.

If you want to preview the development, **you can pull the project down and run it with "F5" in VSCode**. Be warned of bugs and missing features though. They are a lot.

Your tutorial is very interesting. I'm very appreciate you works. The extension is not mature enough. I even found a new bug by watching your video 😂. Deletion should be sync with the Yaml text without any manual actions. I will try to fix it in couple days.

**Comment by DonKolia** [comment]
You can also create a plugin for BQ 2.0 that reads the old conversation format. Addon

**Comment by DonKolia** [comment]
![image](https://github.com/EverMCServer/vscode-BetonQuest/assets/55564649/0b3e7cc7-9328-4045-9941-c71bb71ecfac)

**Comment by k-jiang** [comment]
> ![image](https://user-images.githubusercontent.com/55564649/277088664-0b3e7cc7-9328-4045-9941-c71bb71ecfac.png)

You are missing node.js or npm packages probably. Try to get a LTS from [here](https://nodejs.org/en/download) then do `npm install`.

**Comment by DonKolia** [comment]
![image](https://github.com/EverMCServer/vscode-BetonQuest/assets/55564649/26a13557-064a-490a-99c3-dc52f3e87a28)
Why exist a line NPC (welcome) to NPC (blacksmithReminder)?? 
In the 1.o version too, remove this line please, is a chaos!
![image](https://github.com/EverMCServer/vscode-BetonQuest/assets/55564649/1b73b296-f405-4096-a18f-2d0b270af0ed)

**Comment by k-jiang** [comment]
> ![image](https://user-images.githubusercontent.com/55564649/277113014-26a13557-064a-490a-99c3-dc52f3e87a28.png) Why exist a line NPC (welcome) to NPC (blacksmithReminder)?? In the 1.o version too, remove this line please, is a chaos! ![image](https://user-images.githubusercontent.com/55564649/277113119-1b73b296-f405-4096-a18f-2d0b270af0ed.png)

It represent an "if ... else ..." check in a logic flow: If condition `firstVisit` and / or `!criminal` of "NPC (welcome)" are not met, the NPC will try to show the "NPC (blacksmithReminder)" instead. The line is for you to follow the `else` logic easily.

**Comment by k-jiang** [comment]
If you don't have any more questions about "2.0 compatibility" specifically, I'm going to close this issue.

If you have any other unrelated issues, please open an new one.
